crmprtd/bc_hydro/download.py

```diff
@@ -0,0 +1,70 @@
+# Downloads data from BC hyrdo
+
+import pysftp
+from tempfile import TemporaryDirectory, SpooledTemporaryFile
+import logging
+import os
+import sys
+from zipfile import ZipFile
+from argparse import ArgumentParser
+
+from crmprtd import logging_args, setup_logging, common_auth_arguments
+
+log = logging.getLogger(__name__)
+
+
+def download(username, auth_key, ftp_server, ftp_dir):
+    # Connect FTP server and retrieve file
+    try:
+        sftp = pysftp.Connection(ftp_server, username=username, 
+                                 private_key = auth_key)
+        with TemporaryDirectory() as tmp_dir:
+            sftp.get_r(ftp_dir, tmp_dir)
+            os.chdir(os.path.expanduser(tmp_dir + '/' + ftp_dir))
+
+            with SpooledTemporaryFile(
+                    max_size=int(os.environ.get('CRMPRTD_MAX_CACHE', 2**20)),
+                    mode='r+') as tempfile:
+
+                for filename in os.listdir(os.getcwd()):
+                    name, extension = os.path.splitext(filename)
+                    if extension == '.zip':
+                        zip_file = ZipFile(filename, 'r')
+                        zip_file.extractall()
```

Comment:
You don't actually need to extract the files from the zip file and write them to disk. The `Zipfile` class provides a `read()` method which is a read interface that does the extraction in memory. See [the examples](https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile.open) for how to use it. Then you can basically dispense with `SpooledTemporaryFile` above.